<launch>

  <arg name="model" default="$(find gazebo_ros_motors)/xacro/test.xacro" />

  <param name="robot_description" command="$(find xacro)/xacro $(arg model)" />
  <node  name="urdf_spawner" pkg="gazebo_ros" type="spawn_model" args="-z 1.0 -unpause -urdf -model motor -param robot_description" respawn="false" output="screen" />

  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="false"/>
    <arg name="headless" value="false"/>
    <arg name="recording" value="false"/>
    <arg name="debug" value="false"/>
  </include>

   <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" />
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />
  
  <node name="link" pkg="gazebo_ros_motors" type="link" />
  
 <node name="control" pkg="pid" type="controller" >
      <param name="Kp" value="9.0" />
      <param name="Ki" value="0.001" />
      <param name="Kd" value="0.0001" />
      <param name="upper_limit" value="24" />
      <param name="lower_limit" value="0" />
      <param name="windup_limit" value="50" />
      <param name="cutoff_frequency" value="1/8" />
      <param name="max_loop_frequency" value="100.0" />
      <param name="min_loop_frequency" value="100.0" />
    </node>
    
    
    

</launch>
